services/QuillLMS/client/app/bundles/Connect/components/incorrectSequence/incorrectSequenceContainer.jsx

```diff
@@ -17,35 +17,24 @@ class IncorrectSequencesContainer extends Component {
     const questionTypeLink = questionType === 'sentenceFragments' ? 'sentence-fragments' : 'questions'
     const actionFile = questionType === 'sentenceFragments' ? sentenceFragmentActions : questionActions
     const question = props[questionType].data[props.match.params.questionID]
-    const incorrectSequences = this.getSequences(question)
+    const incorrectSequences = question.incorrectSequences
 
-    this.state = { orderedIds: null, questionType, actionFile, questionTypeLink, incorrectSequences };
-  }
-
-  componentDidMount() {
-    const { actionFile } = this.state
-    const { getUsedSequences } = actionFile
-    const { dispatch, match } = this.props
-    const { params } = match
-    const { questionID } = params
-    if (getUsedSequences) {
-      dispatch(getUsedSequences(questionID))
-    }
-  }
+    const sequencesCollection = hashToCollection(incorrectSequences)
+    const orderedIds = sequencesCollection.sort((a, b) => a.order - b.order).map(b => b.key);
 
-  componentDidUpdate(previousProps, previousState) {
-    const question = this.props[this.state.questionType].data[this.props.match.params.questionID]
-    const incorrectSequences = this.getSequences(question)
-
-    if (!_.isEqual(previousState.incorrectSequences, incorrectSequences)) {
-      this.setState({ incorrectSequences, orderedIds: null });
-    };
+    this.state = { orderedIds: orderedIds, questionType, actionFile, questionTypeLink, incorrectSequences };
   }
 
-  getSequenceIndexByKey = (key) => {
-    const { incorrectSequences, } = this.state
+  UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
It feels like bad practice to add a new `UNSAFE` lifecycle method here. I'm sure this is what we're doing elsewhere, and if we were sharing code it would be one thing, but it's just going to be more work to clean up down the road. I'm okay if we want to ship with this since I know the curriculum team really needs this functionality, but then I think the next card you work on should be a refactor here.